pandas/indexes/numeric.py

```diff
@@ -162,6 +162,72 @@ def _assert_safe_casting(cls, data, subarr):
 Int64Index._add_numeric_methods()
 Int64Index._add_logical_methods()
 
+_uint64_descr_args = dict(
+    klass='UInt64Index',
+    ltype='unsigned integer',
+    dtype='uint64',
+    extra=''
+)
+
+
+class UInt64Index(NumericIndex):
+    __doc__ = _num_index_shared_docs['class_descr'] % _uint64_descr_args
+
+    _typ = 'uint64index'
+    _arrmap = _algos.arrmap_uint64
+    _left_indexer_unique = _join.left_join_indexer_unique_uint64
+    _left_indexer = _join.left_join_indexer_uint64
+    _inner_indexer = _join.inner_join_indexer_uint64
+    _outer_indexer = _join.outer_join_indexer_uint64
+
+    _can_hold_na = False
+
+    _engine_type = _index.UInt64Engine
+
+    _default_dtype = np.uint64
+
+    @property
+    def inferred_type(self):
+        return 'integer'
+
+    @property
+    def asi8(self):
+        # do not cache or you'll create a memory leak
+        return self.values.view('u8')
+
+    @Appender(_index_shared_docs['_convert_scalar_indexer'])
+    def _convert_scalar_indexer(self, key, kind=None):
+        assert kind in ['ix', 'loc', 'getitem', 'iloc', None]
+
+        # don't coerce ilocs to integers
+        if kind != 'iloc':
+            key = self._maybe_cast_indexer(key)
+        return (super(UInt64Index, self)
+                ._convert_scalar_indexer(key, kind=kind))
+
+    def _wrap_joined_index(self, joined, other):
+        name = self.name if self.name == other.name else None
+        return Int64Index(joined, name=name)
+
+    @classmethod
+    def _assert_safe_casting(cls, data, subarr):
+        """
+        Ensure incoming data can be represented as uints.
+        """
+        if not issubclass(data.dtype.type, np.integer):
```

Comment:
Good catch - `np.unsignedinteger` in fact.  I also updated `Int64Index` to use `np.signedinteger`.